src/m-utils/index.js

```diff
@@ -0,0 +1,20 @@
+'use babel'
+
+/**
+ *  Small tools to make intentional mutations easier to read and understand.
+ */
+
+// Replace one array with another while keeping the original pointer
+//
+export const replaceArrayInPlace = (oldArray, newArray) => {
+  oldArray.splice(0, oldArray.length)
+  Array.prototype.push.apply(oldArray, newArray)
+  return oldArray
+}
+
+// Replace 1 item in an array
+//
+export const spliceReplace1 = (elem, array) => {
```

Comment:
🤔

Reply:
It would be nice if I can at least keep the function name from being noticeably longer than the code it contains. 🤣

Reply:
Leave it as is and put it in the description then? Or just leave it alone if you want, just wanted to make sure that you knew that subtlety.

Reply:
> Put it in the description then

Expanding the comment sounds good to me. I hadn't really considered the case of replacing all instances of a given element when I wrote it. I think `spliceReplaceAllOf1` would be a reasonable (if somewhat ugly) name for it's counterpart if for some reason we need to write it in the future.

Reply:
Seems I kind of messed this up all around (even though it is succeeding for its intended use as-is).

For starters, this is a deletion, not a replace, so the name is completely wrong. Whoops. 

But I also forgot to handle the case of trying to delete a non-existent element (I guess because it's not a realistic case as it's currently used, unless we break something at the call-site). If you try to delete an element that doesn't exist, it will delete the last element in the array. 🤦‍♂️